components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2181,6 +2181,37 @@ public ServiceProvider getApplication(int applicationId) throws IdentityApplicat
         }
     }
 
+    @Override
+    public ServiceProvider getApplicationWithRequiredAttributes(int applicationId, List<String> requiredAttributes)
+            throws IdentityApplicationManagementException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+        try {
+            // Load basic application data
+            ServiceProvider serviceProvider = getBasicApplicationData(applicationId, connection);
+            if (CollectionUtils.isNotEmpty(requiredAttributes)) {
+                List<ServiceProviderProperty> propertyList = getServicePropertiesBySpId(connection, applicationId);
+                for (String requiredAttribute : requiredAttributes) {
+                    if (requiredAttribute.equals("advancedConfigurations")) {
+                        readAndSetConfigurationsFromProperties(propertyList,
+                                serviceProvider.getLocalAndOutBoundAuthenticationConfig());
+                        serviceProvider.setSpProperties(propertyList.toArray(new ServiceProviderProperty[0]));
+                        serviceProvider.setCertificateContent(getCertificateContent(propertyList, connection));
+                    }
+                    if (requiredAttribute.equals("templateId")) {
+                        serviceProvider.setTemplateId(getTemplateId(propertyList));
+                    }
+                }
+            }
+            return serviceProvider;
+        } catch (SQLException | CertificateRetrievingException e) {
+            throw new IdentityApplicationManagementException("Failed to gather required attributes for application " +
+                        "with id: " + applicationId, e);
+        } finally {
+            IdentityApplicationManagementUtil.closeConnection(connection);
```

Comment:
2 util classes used to get connection and close connection, is that intentional?

Reply:
Yes. IdentityApplicationManagementUtil.close() calls the IdentityDatabaseUtil.close() and used in getServiceProvider() above.